Description: js/ui/background.js: Use Ubuntu purple to match login screen
 This way the login animation will appear to expand over the login
 screen (system background) instead of suddenly replacing it.
Original author: Didier Roche <didrocks@ubuntu.com>
Author: Daniel van Vugt <daniel.van.vugt@canonical.com>
Bug-Ubuntu: https://launchpad.net/bugs/1813119
Last-Update: 2019-03-01
Forwarded: not-needed

Index: gnome-shell/js/ui/background.js
===================================================================
--- gnome-shell.orig/js/ui/background.js
+++ gnome-shell/js/ui/background.js
@@ -101,6 +101,7 @@ const GnomeDesktop = imports.gi.GnomeDes
 const Meta = imports.gi.Meta;
 const Signals = imports.signals;
 
+const Desktop = imports.misc.desktop;
 const LoginManager = imports.misc.loginManager;
 const Main = imports.ui.main;
 const Params = imports.misc.params;
@@ -489,6 +490,25 @@ let _systemBackground;
 
 var SystemBackground = class SystemBackground {
     constructor() {
+        // Yaru is default ubuntu theme, match GDM look
+        if (Desktop.is("ubuntu")) {
+            if (_systemBackground == null) {
+                _systemBackground = new Meta.Background({ meta_display: global.display });
+                let [, ubuntuColor] = Clutter.Color.from_string('#2C001E');
+                _systemBackground.set_color(ubuntuColor);
+            }
+            this.actor = new Meta.BackgroundActor({ meta_display: global.display,
+                                                    monitor: 0,
+                                                    background: _systemBackground });
+            let id = GLib.idle_add(GLib.PRIORITY_DEFAULT, () => {
+                this.emit('loaded');
+                return GLib.SOURCE_REMOVE;
+            });
+            GLib.Source.set_name_by_id(id, '[gnome-shell] SystemBackground.loaded');
+
+            return;
+        }
+
         let file = Gio.File.new_for_uri('resource:///org/gnome/shell/theme/noise-texture.png');
 
         if (_systemBackground == null) {
